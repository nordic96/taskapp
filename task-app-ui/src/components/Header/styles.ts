import { SxProps } from "@mui/material";

export const HeaderStyles: SxProps = {
    padding: '8px 16px',
    display: 'flex',
    gap: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#24292f',
    height: 70,
};