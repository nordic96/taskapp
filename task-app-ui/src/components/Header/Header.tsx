import React from 'react';

import { AppBar, Box, Typography } from '@mui/material';

import logo from '../../logo.svg';

import { HeaderStyles } from './styles';

const Header = () => {
    return (
        <AppBar id={'task-app-header'} sx={HeaderStyles}>
            <Box display={'flex'} flexDirection={'row'} alignItems={'center'}>
                <img src={logo} style={{ width: 56, height: 56 }} className="App-logo" alt="logo" />
                <Typography sx={{ fontSize: 28, fontWeight: 'bold' }} id={'nav-home'} className={'NavMenu'}>
                    {'Task Management'}
                </Typography>
            </Box>
        </AppBar>
    );
};

export default Header;