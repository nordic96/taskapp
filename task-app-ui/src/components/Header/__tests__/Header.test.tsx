import React from 'react';
import { render } from 'enzyme';
import Header from '../Header';

describe('Testing Header Component', () => {
    it('Should match with snapshot', () => {
        const wrapper = render(<Header />);
        expect(wrapper).toMatchSnapshot(); 
    });
});
