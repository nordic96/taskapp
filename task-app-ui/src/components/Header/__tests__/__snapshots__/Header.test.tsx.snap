// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header Component Should match with snapshot 1`] = `
LoadedCheerio {
  "0": <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-vls1af-MuiPaper-root-MuiAppBar-root"
    id="task-app-header"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 NavMenu css-l6pxfq-MuiTypography-root"
      id="nav-home"
    >
      Task Management
    </p>
  </header>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
